### ISSUE TITLE
WindowsPlugin: support changing codepage

### ISSUE THREAD

**Issue by nigredo-tori** [sbt/sbt-native-packager#1175]
I'm trying to build a Windows installer for my application using `WindowsPlugin` and `JavaPackaging`. However, cyrillic letters in some fields lead to WiX error messages like this one:
```
[info] C:\dev\tmp\snp-test\target\windows\snp-test.wix(3) : error LGHT0311 : A string was provided with characters that are not available in the specified database code page '1252'. Either change these characters to ones that exist in the database's code page, or update the database's code page by modifying one of the following attributes: Product/@Codepage, Module/@Codepage, Patch/@Codepage, PatchCreation/@Codepage, or WixLocalization/@Codepage.
```

As far as I understand, MSI packages have to specify the encoding for the text they use in case it uses a charset other then the default one (Windows-1252). WiX seems to support this, but the corresponding knob is not available in this project. Relevant WiX manual page: http://wixtoolset.org/documentation/manual/v3/overview/codepage.html .

## Expected behaviour

SNP supports changing codepage in generated WiX scripts, allowing fields to contain corresponding characters.

## Actual behaviour

SNP does not support changing codepage in generated WiX scripts, cyrillic letters lead to failing build.

## Information

To reproduce the issue, run `Windows/packageBin` in the following project (SBT 1.2.1, SNP 1.3.14):

```scala
version in ThisBuild := "1.2.3"

enablePlugins(
  JavaAppPackaging,
  WindowsPlugin
)

Compile / mainClass := Some("Main")

maintainer := "Проверка1"
packageSummary := "Проверка2"
packageDescription := "Проверка3"
```

## UPD:

Looking past this, it seems that the autogenerated WiX script is lacking in a few other aspects. I'll probably just roll my own `wixFile`, so this issue is not really urgent.

**Comment by muuki88** [comment]
Thanks for the detailed and well written feature request @nigredo-tori :smiley: 

Windows support was always very community driven as this is rather a rare use case. What you can try is the `JdkPackagerPlugin`: https://sbt-native-packager.readthedocs.io/en/stable/formats/jdkpackager.html
This may solve your issues as it relies on the JDK8 packager tool.

We are happy to merge pull requests fixing or enhancing the windows support :heart:

**Comment by nigredo-tori** [comment]
@muuki88 , thank you for your reply. I've considered using `JdkPackagerPlugin`, but it looks far too rigid for my use case - so I would have to write a custom WiX script anyway. Also, since `javapackager` is no longer in JDK (since Java 11), making it (and Ant) a part of my build is probably not the best solution. There's also `jlink` (which, as I understand, is the `Archetype`-shaped half of `javapackager`), but I am stuck with Java 8 at the moment, and `jlink` is not supported by SNP anyway.

**Comment by muuki88** [comment]
> Also, since javapackager is no longer in JDK (since Java 11), making it (and Ant) a part of my build is probably not the best long-term solution

That is totally correct! I missed that. Oh man. This means we can deprecate that feature

**Comment by muuki88** [comment]
Can you shed some light on the `lightOptions` thing :smile: Is it an sbt or wix setting?

**Comment by nigredo-tori** [comment]
> Can you shed some light on the lightOptions thing :smile: Is it an sbt or wix setting?

I see what you did there :smile:. `lightOptions` is a `WindowsPlugin` setting that allows passing arbitrary options to the `light` utility in WiX (the linker).

Passing `-cultures:ru-ru` using `lightOptions` allowed my WiX build to complete without errors, but I later found out that some of the strings were not displayed correctly in the file properties dialog for the generated MSI. Ultimately, for a non-English MSI one has to specify `-cultures` **as well as** `Product/@Codepage` and `Package/@SummaryCodepage`. So maybe I was too hasty to close this.

**Comment by muuki88** [comment]
Then let's reopen :smiley: 

If you have time and passion to do this, feel free to ask about anything :)
